src/UtilityFunctions.cpp

```diff
@@ -43,8 +46,15 @@ int sumNumbersFromGenerators(std::vector<GeneratorType> const & types, int num)
 {
     // TODO: Write a loop that goes over a collection of GeneratorType values (types)
     // and creates generator for given type (use createGenerator function),
+
+    std::vector<std::unique_ptr<NumberGenerator> > generators;
+    std::for_each(types.begin(), types.end(), [&generators](auto& el) { generators.push_back(std::move(createGenerator(el)));} );
```

Comment:
You can use `generators.emplace_back()` instead of `generators.push_back()`.  It supports moving by default.
Also range-based for loop can be used instead of `std::for_each`

Reply:
thanks for suggestions, I've just pushed simplified version.

Reply:
Looks good 👍